lib/logpipe.c

```diff
@@ -94,3 +95,21 @@ log_pipe_forward_notify(LogPipe *self, gint notify_code, gpointer user_data)
 {
   log_pipe_notify(self->pipe_next, notify_code, user_data);
 }
+
+void
+log_pipe_set_persist_id(LogPipe *self, const gchar *persist_id)
+{
+  if (self->persist_id != NULL)
+    g_free(self->persist_id);
+
+  self->persist_id = persist_id;
+}
+
+const gchar *
+log_pipe_get_persist_id(const LogPipe *self)
+{
+  if (self->generate_persist_id != NULL)
+    return self->generate_persist_id(self);
+  else
+    return self->persist_id;
+}
```

Comment:
I would love to see the common case handled by common code, e.g. when persist_id is in fact specified in the configuration, it would be nice _NOT_ to call the LogPipe specific function. Now all of them has to handle the persist_id != NULL case explicitly, which is very easy to get wrong. If the user specified persist_id it should be used without the potential of messing it up in the driver code.